dotnet/src/Connectors/Connectors.Memory.Pinecone/Model/IndexMetric.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel.Connectors.Memory.Pinecone.Model;
+
+/// <summary>
+/// The vector similarity metric of the index
+/// </summary>
+/// <value>The vector similarity metric of the index</value>
+[JsonConverter(typeof(IndexMetricJsonConverter))]
+public enum IndexMetric
+{
+    /// <summary>
+    /// Enum Euclidean for value: euclidean
+    /// </summary>
+    [EnumMember(Value = "euclidean")]
+    Euclidean = 1,
+
+    /// <summary>
+    /// Enum Cosine for value: cosine
+    /// </summary>
+    [EnumMember(Value = "cosine")]
+    Cosine = 2,
+
+    /// <summary>
+    /// Enum Dotproduct for value: dotproduct
+    /// </summary>
+    [EnumMember(Value = "dotproduct")]
+    Dotproduct = 3
+
+}
+
+public class IndexMetricJsonConverter : JsonConverter<IndexMetric>
```

Comment:
There are three new converters:
- PodTypeJsonConverter
- IndexMetricJsonConverter
- IndexStateJsonConverter

Looks like the implementation is the same in all converters, they check for value in `EnumMember` attribute. A couple of options here:
- Check if standard `JsonStringEnumConverter` covers `EnumMember` attribute, if yes, use it.
- Check for already existing converters in `System.Text.Json` which cover `EnumMember`. If exist, use it.
- Create own converter to cover `EnumMember` attribute but make it generic for all enums.